code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java

```diff
@@ -333,7 +333,7 @@ public void setMapDao(GenericMapDao mapDao) {
                         .and(INSTANCE.REMOVED.isNull())
                         .and(INSTANCE.STATE.in(InstanceConstants.STATE_STARTING, InstanceConstants.STATE_RESTARTING, InstanceConstants.STATE_RUNNING))
                         .and(INSTANCE_HOST_MAP.REMOVED.isNull())
-                        .and(PORT.STATE.in(CommonStatesConstants.ACTIVE, CommonStatesConstants.ACTIVATING))
+                        .and(PORT.STATE.notIn(CommonStatesConstants.REMOVED, CommonStatesConstants.REMOVING, CommonStatesConstants.PURGED))
```

Comment:
Shoot.  I'll add purging in here.  thanks for catching this.  so many states for port :-P